tests/basic/storage_pool_test.go

```diff
@@ -9762,3 +9762,110 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("rebalanceshutdown-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get Pool with running IO on the cluster
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		if IsLocalCluster(*nodeDetail) != true {
+			log.FailOnError(fmt.Errorf("This test will only support onprem vms"), "is this onprem?")
+		}
+		// Get Total Pools present on the Node present
+		poolDetails, err := GetPoolsDetailsOnNode(*nodeDetail)
+		log.FailOnError(err, "Failed to get Pool Details from Node [%v]", nodeDetail.Name)
+		log.InfoD("List of Pools present in the node [%v]", poolDetails)
+		poolToBeResized, err := GetStoragePoolByUUID(poolDetails[0].Uuid)
+		drvSize, err := getPoolDiskSize(poolToBeResized)
+		log.FailOnError(err, "error getting drive size for pool [%s]", poolToBeResized)
+		//getting the eligible pools of the node to initiate expansion
+		eligibility, err := GetPoolExpansionEligibility(nodeDetail)
+		log.FailOnError(err, "error checking node [%s] expansion criteria", nodeDetail.Name)
+		if !eligibility[poolUUID] {
+			log.FailOnError(fmt.Errorf("cannot add drive to the pool selected %s as it is full and not eligible for expansion through drive addition", poolUUID), "cannot add drive into the pool")
+		}
+
+		driveSize := drvSize * 1
+		expectedSize := (poolToBeResized.TotalSize / units.GiB) + driveSize
+		err = Inst().V.ExpandPool(poolToBeResized.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, true)
+		log.FailOnError(err, "error getting drive size for pool [%s]", poolToBeResized)
+		expandedPool, err := GetStoragePoolByUUID(poolToBeResized.Uuid)
+		log.FailOnError(err, "error getting pool by using id %s", poolToBeResized.Uuid)
+		dash.VerifyFatal(expandedPool == nil, false, fmt.Sprintf("Pool selected for expansion is %s(UUID of pool)", expandedPool.Uuid))
+		err = WaitForExpansionToStart(poolToBeResized.Uuid)
+		log.FailOnError(err, "error getting pool by using id %s", poolToBeResized.Uuid)
+		var connect node.ConnectionOpts
+		connect.Timeout = 60
+		connect.TimeBeforeRetry = 10
+		err = Inst().N.ShutdownNode(*nodeDetail, node.ShutdownNodeOpts{
+			Force:          true,
+			ConnectionOpts: connect,
+		})
+		log.FailOnError(err, "failed to shutdown the node %s", nodeDetail.Name)
+		time.Sleep(300 * time.Second)
+		log.InfoD("sleeping for 5 mins to wait for shutdown to be completed")
+		t := func() (interface{}, bool, error) {
+			err = Inst().N.PowerOnVM(*nodeDetail)
+			if err != nil {
+				return nil, false, err
+			}
```

Comment:
here both cases you are returning false, so retry wont happen. please fix it if this is not expected behaviour